### DISCUSSION TITLE
TabNine Support

### DISCUSSION THREAD

**Discussion by MiroloTech** [godotengine/godot-proposals#7693] [Scripting]
I was just wondering how difficult ( and expensive ) it would be to add TabNine ( an AI coding assistant ) support to the Godot Editor and if it would make sense? It feels like such a useful tool.

**Comment by AThousandShips** [comment]
If you are talking about adding a paid feature to the engine that's probably not going to happen, first of all the engine can't contain anything that isn't open source, second of all it's unlikely that money would be spent to add a feature to a free and open source engine, if you want it as a plugin/extension that's a different matter though

**Reply by MiroloTech** [reply]
It was more thought like a free support like VisualStudio has, where you don't have to pay, but don't get the full experience. Also Godot could make its own varient of a TabNine like AI.

**Reply by AThousandShips** [reply]
Thats a massive tasks and has little benefit for the users, better handled as a plugin or extension, especially with the way AI is currently progressing it's unclear how it will develop, it is still very much a fad, and the current evolution is so fast that it's very easy to waste a lot of time on something that is outdated in a year
